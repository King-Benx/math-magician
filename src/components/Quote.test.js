import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

test('the quotes page renders correctly', () => {
  render(<Quote quote="sample text" author="John Doe" />);
  const quoteContent = screen.getByText('sample text');
  const authorContent = screen.getByText('John Doe');
  expect(quoteContent).toBeTruthy();
  expect(authorContent).toBeTruthy();
});

test('the quotes component fully renders', () => {
  const tree = renderer.create(<Quote quote="sample text" author="John Doe" />).toJSON();
  expect(tree).toMatchSnapshot();
});
