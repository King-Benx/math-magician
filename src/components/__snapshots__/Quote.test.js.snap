// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the quotes component fully renders 1`] = `
<div
  className="quote"
>
  <span>
    sample text
  </span>
  <span>
     -- 
  </span>
  <span>
    John Doe
  </span>
</div>
`;
